### PR TITLE
Set var.card

### PR DESCRIPTION
**Pull request by glelouet**
- added SetVarImpl. getCard() and setCard(IntVar)
 - added corresponding simple test
 - specified setVar(String name, int... values) instead  of setVar(String name, int[] value) with null test on values

### CODE REVIEW

File: src/main/java/org/chocosolver/solver/variables/impl/SetVarImpl.java [hunk 1]
```diff
@@ -130,6 +134,29 @@ public ISet getLB() {
 	public ISet getUB() {
 		return ubReadOnly;
 	}
+	
+	@Override
+	public IntVar getCard() {
+		if(cardinality==null){
+			int ubc =  ub.getSize(), lbc = lb.getSize();
```
Comment (cprudhom) [inline src/main/java/org/chocosolver/solver/variables/impl/SetVarImpl.java]: Codacy Issue found: [Use one line for each declaration, it enhances code readability.](https://www.codacy.com/app/cprudhom/choco-solver/file/2534693622/issues/source?bid=3218989&fileBranchId=3284892#l141)

**Comment by jgFages** [comment]
I have two remarks about "specified setVar(String name, int... values) instead of setVar(String name, int[] value) with null test on values"

1) The "value" of a set variable is a set of integers so the method should be named setVar(String name, int... value) (no "s" at the end of value). 

2) I see it is a bit more convenient to avoid writing new int[]{} thanks to the ... but I am not sure it is good. I fear a user mimics integer variables passing 0, 9 and thinking it will represent [0,9] instead of {0,9}, or even worse, all its subsets... What do you think?

**Comment by glelouet** [comment]
1) It depends whether you consider the parameter is a single it array or an array of it, I don't really care. I'm used to using plural form when there are several elements, whether through a collection, an iterable, a stream, an array, which do not have their own behavior.

2) if a user wants to make an intVar as a setVar, we can't help him. SetVar and IntVar are very dissociated and trying to use one like another would surely lead to errors anyhow.
However I like the idea of constructing a setVar over a range (like python), what do you think of adding setVarOver(name, int min, int max) ?

**Comment by jgFages** [comment]
2) Are you suggesting : 
setVarOver(name, int min, int max) returns a constant [min, max]
and 
setVarOver(name, int minLB, int maxLB, int minUB, int maxUB) return a variable?

**Comment by glelouet** [comment]
SetVar IVariableFactory.setVarOver(String name, int min, int max) returns a new SetVar with array of values fixed to {min..max}

**Comment by jgFages** [comment]
good

(I did not want it to create a variable that is a subset of {min...max}. The variable should specify both the LB and the UB)

**Comment by glelouet** [comment]
https://github.com/chocoteam/choco-solver/issues/411 ?

**Comment by jgFages** [comment]
I wait #411 to be over before merging this PR

**Pull request merged**